api/src/mappers.js

```diff
@@ -0,0 +1,94 @@
+const PARAMETERS = {
+  '/items': ['code', 'exact', 'fields', 'fuzzy', 'inclusive', 'name', 'search'],
+};
+
+/**
+ * Parse query parameters.
+ *
+ * @param  {Object}        params    Request query parameters.
+ * @param  {Array<String>} whiteList Valid parameters per implementation.
+ * @return {Object}                  Object containing valid and invalid parameters.
+ */
+export const parseRequest = (request) => {
+  const { url } = request;
+  const whitelist = PARAMETERS[url];
+  const { params } = request;
+  return Object.keys(params).reduce(
+    ({ parameters }, param) => {
+      const { valid: validParameters, invalid: invalidParameters } = parameters;
+      if (whitelist.includes(param))
+        return { parameters: { valid: { ...validParameters, param }, invalid: invalidParameters } };
+      return { parameters: { valid: validParameters, invalid: { ...invalidParameters, param } } };
+    },
+    { parameters: { valid: [], invalid: [] } }
+  );
+};
+
+/**
+ * Map API request to GraphQL+- payload. Query parameters are assumed valid.
+ *
+ * @param  {Object} request REST api request.
+ * @return {String}         GraphQL+- response from Dgraph server.
+ */
+export const mapRequest = (request) => {
+  const { query } = request;
+  const { code, search } = query;
+
+  if (code) {
+    return `{
+      query(func: has(code)) @filter(eq(code, ${code}))
+      {	
+        code
+        description
+      }
+    }`;
+  } else if (search) {
+    return `{
+      query(func: has(code)) @filter(allofterms(description, "${search}")) {
+        code
+        description
+      }
+    }`;
+  } else {
+    // Default to all unit of use entities
+    return `{
+      query(func: has(code)) @filter(eq(type,"unit_of_use")) {
+        code
+        description
+      }
+    }`;
+  }
+};
+
+/**
+ *
+ * Map Dgraph response to valid v1 API JSON payload.
+ *
+ * @param  {Object} request  GraphQL+- request sent to Dgraph server.
```

Comment:
```suggestion

```
Another comment thing - I think your description/naming is definitely better than my original one though ;)

Reply:
Oops. 20940fe6a9416424f8d8ad774e84d7722dbbc614.